src/mapgen/mapgen_fractal.cpp

```diff
@@ -66,13 +67,17 @@ MapgenFractal::MapgenFractal(MapgenFractalParams *params, EmergeManager *emerge)
 	julia_z          = params->julia_z;
 	julia_w          = params->julia_w;
 
-	//// 2D terrain noise
-	noise_seabed       = new Noise(&params->np_seabed, seed, csize.X, csize.Z);
+	//// 2D noise
+	if (spflags & MGFRACTAL_TERRAIN)
+		noise_seabed = new Noise(&params->np_seabed, seed, csize.X, csize.Z);
```

Comment:
Please initialize `noise_seabed` to `nullptr` if this flag is not set. Safety first.
PS: What do you think about renaming the pointer to `noise_terrain` to be like its setting and flag?

Reply:
Ah thanks, i was thinking about your previous advice about nullptr, then forgot.

Will consider, 'seabed' isn't ideal as it actually also creates islands.

Reply:
Well meh, changing the noise name could break worlds, so i'd rather not.

Reply:
Done.

Reply:
> Well meh, changing the noise name could break worlds, so i'd rather not.

No it won't break worlds. This is only a variable name which might even be stripped out by the compiler. What matters for compatibility are values bound to file data: such as settings, old default values, map data or mod configurations. C++ internal names have zero impact on that.

Reply:
Ah i see, you mean without changing the setting name too. That would not break worlds but would be messy and confusing, for no good reason. All noise parameter setting names match the internal variable names.
This noise is best named 'seabed', but the flag is for the underground too so is best named 'terrain' and not 'seabed'. There's no need for the 2 to match.